File: cmd/prism-cli/domain/progress_test.go
package domain

import (
	"path/filepath"
	"testing"
)

func TestNewProgressFile_LegacyFlat(t *testing.T) {
	// Legacy: .prism/stories/stories.json -> .prism/shared/spectrum/progress.md
	storiesPath := filepath.Join("project", ".prism", "stories", "stories.json")
	pf := NewProgressFile(storiesPath)

	expected := filepath.Join("project", ".prism", "shared", "spectrum", "progress.md")
	if pf.Path != expected {
		t.Errorf("legacy path: got %q, want %q", pf.Path, expected)
	}
}

func TestNewProgressFile_EpicScoped(t *testing.T) {
	// Epic: .prism/stories/<epic>/stories.json -> .prism/shared/spectrum/<epic>/progress.md
	storiesPath := filepath.Join("project", ".prism", "stories", "user-auth", "stories.json")
	pf := NewProgressFile(storiesPath)

	expected := filepath.Join("project", ".prism", "shared", "spectrum", "user-auth", "progress.md")
	if pf.Path != expected {
		t.Errorf("epic path: got %q, want %q", pf.Path, expected)
	}
}

func TestNewProgressFile_EpicWithDashes(t *testing.T) {
	storiesPath := filepath.Join("project", ".prism", "stories", "multi-view-dashboard", "stories.json")
	pf := NewProgressFile(storiesPath)

	expected := filepath.Join("project", ".prism", "shared", "spectrum", "multi-view-dashboard", "progress.md")
	if pf.Path != expected {
		t.Errorf("epic with dashes: got %q, want %q", pf.Path, expected)
	}
}

func TestNewProgressFile_AbsolutePath_Legacy(t *testing.T) {
	// Test with absolute-style path
	storiesPath := filepath.Join(string(filepath.Separator), "home", "user", "project", ".prism", "stories", "stories.json")
	pf := NewProgressFile(storiesPath)

	expected := filepath.Join(string(filepath.Separator), "home", "user", "project", ".prism", "shared", "spectrum", "progress.md")
	if pf.Path != expected {
		t.Errorf("absolute legacy: got %q, want %q", pf.Path, expected)
	}
}

func TestNewProgressFile_AbsolutePath_Epic(t *testing.T) {
	storiesPath := filepath.Join(string(filepath.Separator), "home", "user", "project", ".prism", "stories", "my-epic", "stories.json")
	pf := NewProgressFile(storiesPath)

	expected := filepath.Join(string(filepath.Separator), "home", "user", "project", ".prism", "shared", "spectrum", "my-epic", "progress.md")
	if pf.Path != expected {
		t.Errorf("absolute epic: got %q, want %q", pf.Path, expected)
	}
}

File: apps/prism-vscode/src/prism/__tests__/progress.test.ts
/**
 * Unit tests for src/prism/progress.ts
 * Ports cmd/prism-cli/domain/progress_test.go
 */

import * as path from "path"
import { progressPathFromStories } from "@prism-core/prism/progress"

// ---------------------------------------------------------------------------
// progressPathFromStories — mirrors TestNewProgressFile_* in Go
// ---------------------------------------------------------------------------

describe("progressPathFromStories", () => {
  test("legacy flat: .prism/stories/stories.json → .prism/shared/spectrum/progress.md", () => {
    const storiesPath = path.join("project", ".prism", "stories", "stories.json")
    const result = progressPathFromStories(storiesPath)
    const expected = path.join("project", ".prism", "shared", "spectrum", "progress.md")
    expect(result).toBe(expected)
  })

  test("epic-scoped: .prism/stories/<epic>/stories.json → .prism/shared/spectrum/<epic>/progress.md", () => {
    const storiesPath = path.join("project", ".prism", "stories", "user-auth", "stories.json")
    const result = progressPathFromStories(storiesPath)
    const expected = path.join(
      "project",
      ".prism",
      "shared",
      "spectrum",
      "user-auth",
      "progress.md",
    )
    expect(result).toBe(expected)
  })

  test("epic with dashes in name", () => {
    const storiesPath = path.join(
      "project",
      ".prism",
      "stories",
      "multi-view-dashboard",
      "stories.json",
    )
    const result = progressPathFromStories(storiesPath)
    const expected = path.join(
      "project",
      ".prism",
      "shared",
      "spectrum",
      "multi-view-dashboard",
      "progress.md",
    )
    expect(result).toBe(expected)
  })

  test("absolute path legacy", () => {
    const storiesPath = path.join(
      path.sep,
      "home",
      "user",
      "project",
      ".prism",
      "stories",
      "stories.json",
    )
    const result = progressPathFromStories(storiesPath)
    const expected = path.join(
      path.sep,
      "home",
      "user",
      "project",
      ".prism",
      "shared",
      "spectrum",
      "progress.md",
    )
    expect(result).toBe(expected)
  })

  test("absolute path epic-scoped", () => {
    const storiesPath = path.join(
      path.sep,
      "home",
      "user",
      "project",
      ".prism",
      "stories",
      "my-epic",
      "stories.json",
    )
    const result = progressPathFromStories(storiesPath)
    const expected = path.join(
      path.sep,
      "home",
      "user",
      "project",
      ".prism",
      "shared",
      "spectrum",
      "my-epic",
      "progress.md",
    )
    expect(result).toBe(expected)
  })
})
